src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityProcessingArray.java

```diff
@@ -115,27 +120,35 @@ public void invalidate() {
         /**
          * Checks if a provided Recipe Map is valid to be used in the processing array
          * Will filter out anything in the config blacklist, and also any non-singleblock machines
+         *
          * @param recipeMap The recipeMap to check
          * @return {@code true} if the provided recipeMap is valid for use
          */
         @Override
         public boolean isRecipeMapValid(RecipeMap<?> recipeMap) {
-            if (GTUtility.findMachineInBlacklist(recipeMap.getUnlocalizedName(), ((IMachineHatchMultiblock) metaTileEntity).getBlacklist()))
+            if (recipeMap == null || GTUtility.findMachineInBlacklist(recipeMap.getUnlocalizedName(), ((IMachineHatchMultiblock) metaTileEntity).getBlacklist()))
                 return false;
 
             return GTUtility.isMachineValidForMachineHatch(currentMachineStack, ((IMachineHatchMultiblock) metaTileEntity).getBlacklist());
         }
 
         @Override
-        protected Recipe findRecipe(long maxVoltage, IItemHandlerModifiable inputs, IMultipleTankHandler fluidInputs, MatchingMode mode) {
-
-            findMachineStack();
-
-            if (currentMachineStack.isEmpty() || this.activeRecipeMap == null)
-                return null;
+        protected boolean shouldSearchForRecipes() {
+            return canWorkWithMachines() && super.shouldSearchForRecipes();
+        }
 
-            //Find available recipes based on the tier of the machines the processing array is operating upon and the voltage provided to the Processing Array
-            return super.findRecipe(maxVoltage, inputs, fluidInputs, mode);
+        public boolean canWorkWithMachines() {
+            if (machineChanged) {
+                findMachineStack();
+                machineChanged = false;
+                previousRecipe = null;
+                if (isDistinct()) {
+                    invalidatedInputList.clear();
+                } else {
+                    invalidInputsForRecipes = false;
+                }
+                return (currentMachineStack.isEmpty() || this.activeRecipeMap == null);
```

Comment:
This should return `!(currentMachineStack.isEmpty() || this.activeRecipeMap == null)` right? We can work with the machines if the new machine stack actually has machine and a valid recipeMap

Reply:
I think this return statement can be outright removed, and the one in the else statement could be always returned instead, at the end of the method.